DEVGUIDE.md

```diff
@@ -0,0 +1,30 @@
+# Developer's Guide to Service-Catalog
+
+## Building
+
+To build the service-catalog you have two options:
+* `make build`
+* `DOCKER=1 make build`
+
+Both will build all of the executables, into the `bin` directory. However,
+the second option will do the build within a Docker container - meaning you
+do not need to have all of the necessary tooling installed on your host
+(such as a golang compiler or glide). Whichever option you choose, the
+results should be the same.
```

Comment:
I worry that on a mac, DOCKER=1 will build Linux binaries (which is fine) but will overwrite any binaries built without DOCKER=1 (which doesn't seem right).